### DISCUSSION TITLE
vm-ebs vs. vm-ami vs. fs on ec2 have different results?

### DISCUSSION THREAD

**Discussion by chrismo** [aquasecurity/trivy#4111] [Q&A]
I ran `trivy` 0.40.0 against an AMI, its underlying EBS snapshot, and then on an EC2 instance made from the AMI, and had different results. Not all of the same Ubuntu 20.04 issues were found (EBS snapshot found the most), and several Go language results were missing from ami and ec2 scans. 

I can provide more details if necessary, and I know the `vm` features are still marked experimental, so perhaps there's a simple explanation - maybe trivy has complete access to the file system in the `ebs` use case?

**Comment by knqyf263** [comment]
@masahiro331 Take a look, please

**Comment by masahiro331** [comment]
@chrismo 
Could you provide me with the AMI ID?
